presto-memory/src/test/java/io/prestosql/plugin/memory/TestMemorySmoke.java

```diff
@@ -162,6 +162,63 @@ public void testJoinDynamicFilteringBlockProbeSide()
                 ImmutableSet.of(1, ORDERS_COUNT, PART_COUNT));
     }
 
+    @Test
+    public void testSemiJoinDynamicFilteringNone()
+    {
+        // Probe-side is not scanned at all, due to dynamic filtering:
+        assertDynamicFiltering(
+                "SELECT * FROM lineitem WHERE lineitem.orderkey IN (SELECT orders.orderkey FROM orders WHERE orders.totalprice < 0)",
+                withBroadcastJoin(),
+                0,
+                ImmutableSet.of(0, ORDERS_COUNT));
+    }
+
+    @Test
+    public void testSemiJoinLargeBuildSideNoDynamicFiltering()
+    {
+        // Probe-side is fully scanned because the build-side is too large for dynamic filtering:
+        assertDynamicFiltering(
+                "SELECT * FROM lineitem WHERE lineitem.orderkey IN (SELECT orders.orderkey FROM orders)",
+                withBroadcastJoin(),
+                toIntExact(LINEITEM_COUNT),
+                ImmutableSet.of(LINEITEM_COUNT, ORDERS_COUNT));
+    }
+
+    @Test
+    public void testPartitionedSemiJoinNoDynamicFiltering()
+    {
+        // Probe-side is fully scanned, because local dynamic filtering does not work for partitioned joins:
+        assertDynamicFiltering(
+                "SELECT * FROM lineitem WHERE lineitem.orderkey IN (SELECT orders.orderkey FROM orders WHERE orders.totalprice < 0)",
+                withPartitionedJoin(),
+                0,
+                ImmutableSet.of(LINEITEM_COUNT, ORDERS_COUNT));
+    }
+
+    @Test
+    public void testSemiJoinDynamicFilteringSingleValue()
+    {
+        assertQueryResult("SELECT orderkey FROM orders WHERE comment = 'nstructions sleep furiously among '", 1L);
```

Comment:
no need to check it twice, it's already checked as part of `testJoinDynamicFilteringSingleValue`